src/coreclr/build.cmd

```diff
@@ -1010,9 +1021,18 @@ REM === Helper routines
 REM ===
 REM =========================================================================================
 
-:CrossgenFailure
+
+REM =========================================================================================
+REM === These two routines are intended for the exit code to propagate to the parent process
+REM === Like MSBuild or Powershell. If we directly exit /b 1 from within a if statement in
+REM === any of the routines, the exit code is not propagated.
```

Comment:
This is another great item / design principal that we could include in scripting design guideline documents.

Reply:
Do we have a doc for those guidelines yet? If not, do we have an issue?